### PR TITLE
Adding cache info file to archive, which includes stack ID

### PR DESCRIPTION
**Pull request by lpusok**

### CODE REVIEW

File: bitrise.yml [hunk 1]
```diff
@@ -373,6 +373,56 @@ workflows:
               exit 1
             fi
 
+  stack-versions:
+    description: |
+      Test that archive_info.json containing stack ID is created.
+    envs:
+    - TMP_DIR: $ORIG_BITRISE_SOURCE_DIR/_tmp_stack_verions
```
Comment (trapacska) [inline bitrise.yml]: typo

File: bitrise.yml [hunk 2]
```diff
@@ -373,6 +373,56 @@ workflows:
               exit 1
             fi
 
+  stack-versions:
```
Comment (trapacska) [inline bitrise.yml]: please add a prefix `test-`, thisworkflow name is not describing what will happen

File: bitrise.yml [hunk 3]
```diff
@@ -373,6 +373,56 @@ workflows:
...
+    - script:
+        title: Check that archive_info.json exists
+        inputs:
+        - content: |-
+            #!/bin/env bash
+            set -ex
+            # remove TMP_DIR to cleanup
+            rm -rf $TMP_DIR
+            # extract the archive
+            firstfile="$(tar -b 1024 -tf /tmp/cache-archive.tar | while read item ; do [[ "$item" =~ .* ]] && echo "$item" && break ; done )"
```
Comment (trapacska) [inline bitrise.yml]: very nice 👍

File: main.go [hunk 4]
```diff
@@ -21,6 +21,7 @@ import (
 const (
 	cacheInfoFilePath = "/tmp/cache-info.json"
```
Comment (trapacska) [inline main.go]: NOTE: oops, /tmp is a different partition then the os default temp dir, so creating and using temp files from this dir might will be a lot slower

File: stack_versions.go [hunk 5]
```diff
@@ -0,0 +1,19 @@
+package main
+
+import (
+	"encoding/json"
+	"fmt"
+)
+
+func stackVersions(stackID string) ([]byte, error) {
```
Comment (trapacska) [inline stack_versions.go]: this name is misleading, there is no version**S**, this function will wrap stack id string in a json format and returns that content in a byte array. I would name this function as: `stackVersionInfo`

Comment (trapacska) [reply stack_versions.go]: or to keep it more generic: archiveInfo -> which then will be stored in an archive_info.json file 👍

File: main.go (hunk 4 above)
Comment (lpusok) [reply main.go]: What do you suggest?

File: main.go [hunk 6]
```diff
@@ -21,6 +21,7 @@ import (
...
 	cacheArchivePath  = "/tmp/cache-archive.tar"
+	stackVersionsPath = "archive_info.json"
```
Comment (godrei) [inline main.go]: NOTE: why not in the /tmp dir?

File: cache_archive.go [hunk 7]
```diff
@@ -112,19 +112,24 @@ func (a *Archive) WriteHeader(descriptor map[string]string, descriptorPth string
 		return err
 	}
 
-	closingHeader := &tar.Header{
+	return a.writeData(b, descriptorPth)
+}
+
+// writeData writes the byte array into the archive.
+func (a *Archive) writeData(data []byte, descriptorPth string) error {
```
Comment (trapacska) [inline cache_archive.go]: writeData is misleading, as it contains predefined tar header, so with this method it is only possible to write regular file into the archive. maybe it makes no other sense but it is great to know that there are many other types than can be included in an archive:
https://golang.org/pkg/archive/tar/#pkg-constants

Comment (trapacska) [reply cache_archive.go]: so handle this comment as a note

**Pull request merged**